client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +48,170 @@ export default class UCIMap extends PureComponent {
         selected_acronym: '',
         eventsInCalendar: AppStore.getEventsInCalendar(),
         currentScheduleIndex: AppStore.getCurrentScheduleIndex(),
+        poly: [],
+        info_markers: [],
+        info_marker: null,
+    };
+
+    getRoute = (day) => {
+        if (day) {
```

Comment:
This fixes the bug where the previous route doesn't get cleared when there's only one class on the map
```suggestion
        // Clear any existing route on the map
        this.setState({ poly: [], info_marker: null });
        
        if (day) {
```